File: ios/CBCoreTests/BoundedSetTests.swift
// Copyright (c) 2017-2019 Coinbase Inc. See LICENSE

import XCTest
@testable import CBCore

class BoundedSetTests: XCTestCase {
    func testBoundedSetMaxSize() {
        let set = BoundedSet<String>(maxSize: 2)

        set.add("hish")
        set.add("johnny")
        set.add("andrew")

        XCTAssertEqual(2, set.count)
        XCTAssertTrue(set.has("johnny"))
        XCTAssertTrue(set.has("andrew"))
        XCTAssertFalse(set.has("hish"))
    }

    func testBoundedItemsTouched() {
        let set = BoundedSet<String>(maxSize: 2)

        set.add("hish")
        set.add("johnny")
        set.add("hish")
        set.add("andrew")

        XCTAssertEqual(2, set.count)
        XCTAssertTrue(set.has("andrew"))
        XCTAssertFalse(set.has("johnny"))
        XCTAssertTrue(set.has("hish"))
    }
}

File: android/core/src/test/java/com/coinbase/wallet/core/util/BoundedSetTests.kt
package com.coinbase.wallet.core.util

import org.junit.Assert.assertEquals
import org.junit.Assert.assertTrue
import org.junit.Assert.assertFalse
import org.junit.Test

class BoundedSetTests {
    @Test
    fun testBoundedSetMaxSize() {
        val set = BoundedSet<String>(2)

        set.add("hish")
        set.add("johnny")
        set.add("andrew")

        assertEquals(2, set.count())
        assertTrue(set.has("johnny"))
        assertTrue(set.has("andrew"))
        assertFalse(set.has("hish"))
    }

    @Test
    fun testBoundedItemsTouched() {
        val set = BoundedSet<String>(2)

        set.add("hish")
        set.add("johnny")
        set.add("hish")
        set.add("andrew")

        assertEquals(2, set.count())
        assertTrue(set.has("andrew"))
        assertFalse(set.has("johnny"))
        assertTrue(set.has("hish"))
    }
}
